Incompatibility with flask 2.3

The [relase notes for flask 2.3](https://flask.palletsprojects.com/en/2.3.x/changes/) contain this:
> json_encoder and json_decoder attributes on app and blueprint, and the corresponding json.JSONEncoder and JSONDecoder classes, are removed.

There is [some more information in the docs for 2.2](https://flask.palletsprojects.com/en/2.2.x/api/#flask.json.JSONEncoder) about `JSONEncoder`.

Currently `JSONEncoder` is used [here in flasgger](https://github.com/flasgger/flasgger/blob/master/flasgger/base.py#L898), leading to this import error:
```
my_api/__init__.py:9: in <module>
    from flasgger import Swagger
/usr/local/lib/python3.10/site-packages/flasgger/__init__.py:10: in <module>
    from .base import Swagger, Flasgger, NO_SANITIZER, BR_SANITIZER, MK_SANITIZER, LazyJSONEncoder  # noqa
/usr/local/lib/python3.10/site-packages/flasgger/base.py:28: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/usr/local/lib/python3.10/site-packages/flask/json/__init__.py)
```

I think the fix would be to port the serialization behavior to [JSONProvider](https://flask.palletsprojects.com/en/2.3.x/api/#flask.json.provider.JSONProvider).

There is some information in the [PR that introduced JSONProvider](https://github.com/pallets/flask/pull/4692) about why they changed this.

--- a/flasgger/base.py
+++ b/flasgger/base.py
@@ -17,7 +17,6 @@ except ImportError:
 from functools import wraps, partial
 from collections import defaultdict
 from flask import Blueprint
-from flask import Markup
 from flask import current_app
 from flask import jsonify, Response
 from flask import redirect
@@ -31,6 +30,7 @@ try:
 except ImportError:
     RequestParser = None
 import jsonschema
+from markupsafe import Markup
 from mistune import markdown
 from .constants import OPTIONAL_FIELDS, OPTIONAL_OAS3_FIELDS
 from .utils import LazyString

--- a/flasgger/base.py
+++ b/flasgger/base.py
@@ -25,7 +25,7 @@ from flask import render_template
 from flask import request, url_for
 from flask import abort
 from flask.views import MethodView
-from flask.json import JSONEncoder
+from flask.json.provider import DefaultJSONProvider
 try:
     from flask_restful.reqparse import RequestParser
 except ImportError:
@@ -895,7 +895,7 @@ class Swagger(object):
 Flasgger = Swagger  # noqa
 
 
-class LazyJSONEncoder(JSONEncoder):
+class LazyJSONEncoder(DefaultJSONProvider):
     def default(self, obj):
         if isinstance(obj, LazyString):
             return str(obj)

--- a/flasgger/base.py
+++ b/flasgger/base.py
@@ -24,13 +24,19 @@ from flask import render_template
 from flask import request, url_for
 from flask import abort
 from flask.views import MethodView
-from flask.json.provider import DefaultJSONProvider
+try:
+    from flask.json.provider import DefaultJSONProvider
+except ImportError:
+    from flask.json import JSONEncoder as DefaultJSONProvider
 try:
     from flask_restful.reqparse import RequestParser
 except ImportError:
     RequestParser = None
 import jsonschema
-from markupsafe import Markup
+try:
+    from markupsafe import Markup
+except ImportError:
+    from flask import Markup
 from mistune import markdown
 from .constants import OPTIONAL_FIELDS, OPTIONAL_OAS3_FIELDS
 from .utils import LazyString

